ntbs-service/wwwroot/source/Components/CascadingDropdown.ts

```diff
@@ -0,0 +1,102 @@
+import Vue from "vue";
+import { getHeaders } from "../helpers";
+import axios from "axios";
+
+type OptionValue = {
+    value: string,
+    text: string,
+    group: string
+};
+
+const DEFAULT_SELECT_INNER_HTML = "<option value=\"\">Please select</option>";
+
+/*
+ Triple filtering dropdown based off FilteringDropdown.ts
+ 
+ This is used in TransferRequest.cshtml currently where the first dropdown filters the next (and resets the 3rd dropdown), and the second dropdown
+ filters the third dropdown.
+
+*/
+const CascadingDropdown = Vue.extend({
+    props: {
+        filterFirstHandlerPath: {
+            type: String
+        },
+        filterSecondHandlerPath: {
+            type: String
+        },
+        filteringRefs: {
+            type: Array
+        }
+    },
+    mounted: function () {
+        this.filteringMounted();
+    },
+    methods: {
+        filteringMounted: function () {
+            this.fetchFilteredList("", this.filterSecondHandlerPath);
+            this.$refs["firstFilterContainer"].classList.remove("hidden");

```

Comment:
If we're leaving this as a 'generic' component, I think the behaviour of this should be configured via a prop. We could leave this up for if we need it to be generic - but at that point we may aswell call this notificationTrasnferCascadingDropdown or similar.